src/chainrules.jl

```diff
@@ -0,0 +1,63 @@
+@non_differentiable validate_woodbury_arguments(A, D, S)
+
+# Rule for Woodbury * Real. 
+# Ignoring Complex version for now. 
+# NOTE: # Can't use Tangent{WoodburyPDMat} here. ChainRules is fine, but Zygote has issues and passes around Tangent{Any}
```

Comment:
does this comment still apply? Does it refer to returning Tangent from the pullback, or dispatching on the argument of the pullback?

Reply:
I didn't dig into this too much, but this is what I did. 

I was checking with an integration test with Zygote like the following:

```
@testset "Zygote Integration" begin
    m = randn(size(A, 1))
    α = 2.1
    test_ad(randn(), R, α, m, A, D, S, x) do R, α, m, A, D, S, x
        W = R * WoodburyPDMat(A, D, S)
        return logpdf(Distributions.GenericMvTDist(α, m, W), x)
    end
end
```

(this is entirely similar to what is going on [here](https://github.com/invenia/PDMatsExtras.jl/blob/master/test/woodbury_pd_mat.jl#L71). 

And basically, with the code-as-is this is fine. Without it, , i.e. with:

```
function _times_pullback(Ȳ::Tangent, A::T, B::Real, proj)  where {T<:WoodburyPDMat} ->
function _times_pullback(Ȳ::Tangent{WoodburyPDMat}, A::T, B::Real, proj)  where {T<:WoodburyPDMat}
```

I get: 
```
ERROR: MethodError: no method matching _times_pullback(::Tangent{Any, NamedTuple{(:A, :D, :S), Tuple{Matrix{Float64}, Diagonal{Float64, Vector{Float64}}, Diagonal{Float64, Vector{Float64}}}}}, ::Float64, ::WoodburyPDMat{Float64, Matrix{Float64}, Diagonal{Float64, Vector{Float64}}, Diagonal{Float64, Vector{Float64}}}, ::NamedTuple{(:A, :B), Tuple{ProjectTo{Float64, NamedTuple{(), Tuple{}}}, ProjectTo{WoodburyPDMat{Float64, Matrix{Float64}, Diagonal{Float64, Vector{Float64}}, Diagonal{Float64, Vector{Float64}}}, NamedTuple{(:A, :D, :S), Tuple{Matrix{Float64}, Diagonal{Float64, Vector{Float64}}, Diagonal{Float64, Vector{Float64}}}}}}})
Closest candidates are:
  _times_pullback(::Tangent, ::Any, ::Any, ::Any, ::Any) at ...
```

I'll note I don't see any instances of the tangent type being specialised in ChainRules (or in BlockDiagonals)

Reply:
oh, I see, it's probably this line: https://github.com/FluxML/Zygote.jl/blob/7f2b16987dc4eab236aabd889444cabdcc6f9caf/src/compiler/chainrules.jl#L128

But anyway, I do agree we don't need to parameterise the `Tangent`. I was just wondering whether we should remove the comment (at least in the present form I found it confusing rather than illuminating)